lens/history.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package lens
+
+import (
+	"context"
+
+	"github.com/ipfs/go-datastore/query"
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/core"
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+// schemaHistoryLink represents an item in a particular schema's history, it
+// links to the previous and next version items if they exist.
+type schemaHistoryLink struct {
+	// The schema version id of this history item.
+	schemaVersionID string
+
+	// The history link to the next schema version, if there is one
+	// (for the most recent schema version this will be None).
+	next immutable.Option[*schemaHistoryLink]
+
+	// The history link to the previous schema version, if there is
+	// one (for the initial schema version this will be None).
+	previous immutable.Option[*schemaHistoryLink]
+}
+
+// targetedSchemaHistoryLink represents an item in a particular schema's history, it
+// links to the previous and next version items if they exist.
```

Comment:
todo: the description is the same as for `schemaHistoryLink`. Would be great to see the difference (except for additional field) just by reading the doc

Reply:
you got me wrong. I didn't mean that one doc should refer to another. 
What I mean is that the 2 structs are very similar but different for a reason.
From a readers perspective the reason is not quite clear just from the docs because the first paragraph the describes them is the same.

Reply:
This (original) comment is actually incorrect. The difference is already noted in the second paragraph, and is really quite visible when reading the docs.